File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/dependencyinjection/ApplicationComponent.java
package com.example.sampleapplicationforgm.dependencyinjection;

import com.example.myfeature.MyFeatureModule;
import com.example.networkutil.NetworkUtilModule;
import com.example.sampleapplicationforgm.application.BasicAppApplication;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class,
        ApplicationModule.class,
        AndroidViewModule.class,
        MyFeatureModule.class,
        NetworkUtilModule.class})
public interface ApplicationComponent extends AndroidInjector<BasicAppApplication> {

    void inject(BasicAppApplication basicAppApplication);

}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/dependencyinjection/ApplicationComponent.kt
package com.example.sampleapplicationforgm.dependencyinjection

import com.example.myfeature.MyFeatureModule
import com.example.networkutil.NetworkUtilModule
import com.example.sampleapplicationforgm.application.BasicAppApplication
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class,
                      ApplicationModule::class,
                      AndroidViewModule::class,
                      MyFeatureModule::class,
                      NetworkUtilModule::class])
interface ApplicationComponent : AndroidInjector<BasicAppApplication?> {
    override fun inject(basicAppApplication: BasicAppApplication?)
}